### ISSUE TITLE
[java] Display replacement string in message of UnnecessaryFullyQualifiedName

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5585]
**Affects PMD Version:**  7.11.0

**Rule:** https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname

**Description:**

Given:
```java
public sealed class Configuration permits app.licensed.common.util.Configuration.MergedConfiguration
{
  public static final class MergedConfiguration
  {}
}
```

without an import statement for `app.licensed.common.util.Configuration.MergedConfiguration`.

I get `[WARNING] PMD Failure: app.licensed.common.util.Configuration:28 Rule:UnnecessaryFullyQualifiedName Priority:4 Unnecessary qualifier 'app.licensed.common.util': 'Configuration' is already in scope because it is declared in an enclosing type.` but this is a false-positive because removing the qualifier results in a compiler error. There are only two legal ways to refer to this class: using its fully-qualified name or an `import` statement.

**Expected outcome:**

No warning

**Actual outcome:**
`PMD Failure: app.licensed.common.util.Configuration:28 Rule:UnnecessaryFullyQualifiedName Priority:4 Unnecessary qualifier 'app.licensed.common.util': 'Configuration' is already in scope because it is declared in an enclosing type.`

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven plugin 3.26.0

**Comment by adangel** [comment]
Note, since MergedConfiguration is a inner class, you don't need to add it explicitly with the permits clause: If there is no permits clause for a sealed class, then automatically all inner classes are taken. You also need to make sure, to extend Configuration by your inner class to have some compilable example.

But yes, if you want to explicitly specify the permits clause, then there is no other way than to use a fully qualified name. So, that's indeed a bug in the rule. Thanks for reporting!

**Comment by cowwoc** [comment]
@adangel Thanks for the clarification about the permits clause. I wasn't aware of that...

**Comment by oowekyala** [comment]
> There are only two legal ways to refer to this class: using its fully-qualified name or an import statement.

I tried this with the following sample:
```java
package foo;
sealed class Foo permits Foo.Inner {
  final static class Inner extends Foo {}
}
```
and this compiles fine with javac 17. This means that `Foo` can indeed be referred to in the permits clause without writing `foo.Foo`. What is the version of your compiler? Is it possible that the problem is not with javac/pmd but eg with intellij showing an error when it shouldn't?

**Comment by cowwoc** [comment]
> To clarify, you cannot write permits MergedConfiguration but you can write permits Configuration.MergedConfiguration. The error message of the rule also says that only the package name app.licensed.common.util is unnecessary, not the full app.licensed.common.util.Configuration. So AFAICS there is no problem here.

You're right. I'm just so used to interpreting `Unnecessary qualifier` as being able to remove everything except for the last term...

Is it possible to make the error message clearer? Instead of:

> PMD Failure: app.licensed.common.util.Configuration:28 Rule:UnnecessaryFullyQualifiedName Priority:4 Unnecessary qualifier 'app.licensed.common.util': 'Configuration' is already in scope because it is declared in an enclosing type.

Can we return this?

> PMD Failure: app.licensed.common.util.Configuration:28 Rule:UnnecessaryFullyQualifiedName Priority:4 Unnecessary qualifier 'app.licensed.common.util': 'Configuration' is already in scope from an enlosing type. Use 'Configuration.MergedConfiguration' instead of 'app.licensed.common.util.Configuration.MergedConfiguration'.

**Comment by oowekyala** [comment]
Yes we could improve that. Thanks for the suggestion.
